### PR TITLE
Tools #2 Homework

### PR DESCRIPTION
**Pull request by kszytko**

### CODE REVIEW

File: src/functions.cpp [hunk 1]
```diff
@@ -14,13 +13,7 @@ void sort(std::vector<int> & unsorted) {
...
 }
 
-int min(std::vector<int> const & values) {
-    int minimum = std::numeric_limits<int>::max();
-    for (const auto & value : values) {
-        if (value < minimum) {
-            minimum = value;
-        }
-    }
-    return minimum;
```
Comment (Kasperice) [inline src/functions.cpp]: Why did you delete min function? It should remain here (as a part of master branch)

**Review by Kasperice — CHANGES_REQUESTED** [review]

File: src/functions.cpp (hunk 1 above)
Comment (kszytko) [reply src/functions.cpp]: My mistake!

File: tutor [hunk 2]
```diff
@@ -0,0 +1,995 @@
...
+
+  1. Przenieś kursor do linii poniżej oznaczonej --->.
+
+  2. By poprawić błędy, naprowadź kursor na znak do usunięcia.
+
+  3. Wciśnij  x  aby usunąć niechciany znak.
+
+  4. Powtarzaj kroki 2. do 4. dopóki zdanie nie jest poprawne.
+
+---> Kkrowa prrzeskoczyła prrzez ksiiężycc.
```
Comment (ziobron) [inline tutor]: hmm... not completed

File: tutor [hunk 3]
```diff
@@ -0,0 +1,995 @@
...
+
+  2. Aby poprawić pierwszy wiersz, ustaw kursor na pierwszym znaku PO tym,
+     gdzie tekst ma być wstawiony.
+
+  3. Wciśnij  i  a następnie wpisz konieczne poprawki.
+
+  4. Po poprawieniu błędu wciśnij <ESC>, by wrócić do trybu Normal.
+     Powtarzaj kroki 2. do 4., aby poprawić całe zdanie.
+
+---> W tej brkje trochę .
```
Comment (ziobron) [inline tutor]: hmm... not completed

File: tutor [hunk 4]
```diff
@@ -0,0 +1,995 @@
...
+
+  3. Kiedy tekst został dodany, wciśnij <ESC> i wróć do trybu Normalnego.
+
+  4. Przenieś kursor do drugiej linii oznaczonej ---> i powtórz kroki 2. i 3.,
+     aby poprawić zdanie.
+
+---> Brakuje tu tro
+     Brakuje tu trochę tekstu.
+---> Tu też trochę bra
+     Tu też trochę brakuje.
```
Comment (ziobron) [inline tutor]: not completed

**Review by ziobron — CHANGES_REQUESTED** [review]
Looks that vimtutor is not completed. I want to grant you points for it so please deliver it. Branch structure looks good.

**Review by DamiXioR — COMMENTED** [review]
nice work

File: src/functions.cpp [hunk 5]
```diff
@@ -1,10 +1,9 @@
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
-#include <algorithm>
+#include "functions.hpp"
 #include <iostream>
-#include <limits>
+#include <algorithm>
```
Comment (Marcinati) [inline src/functions.cpp]: Please sort headers.

File: tutor (hunk 2 above)
Comment (kszytko) [reply tutor]: Found that right file has been saved in different place.

**Comment by ziobron** [comment]
+4 XP Narzędzia#2 – vimtutor
+5 XP Narzędzia#2 – git - konflikty
+4 XP Narzędzia#2 – flagi kompilacji
+3 XP Narzędzia#2 – git - cherrypick
+8 XP delivery before 07.06.2020

{ "requirement_ids": [1119, 1999, 1119, 2000, 1119, 2001, 1119, 2002], "points": [2, 4, 2, 5, 2, 4, 2, 3] }
